internal/senders/beanstlakd_sender.go

```diff
@@ -0,0 +1,43 @@
+package senders
+
+import (
+  "encoding/json"
+  models "github.com/shoplineapp/captin/models"
+  beanstalk "github.com/beanstalkd/go-beanstalk"
+  "fmt"
+  "time"
+  log "github.com/sirupsen/logrus"
+)
+
+
+// BeanstalkdSender - Send Event to beanstalkd
+type BeanstalkdSender struct{}
+
+// SendEvent - #BeanstalkdSender SendEvent
+func (c *BeanstalkdSender) SendEvent(e models.IncomingEvent, d models.Destination) error {
+  host := fmt.Sprintf("%s:%s", e.Control["beanstalkd_host"], e.Control["beanstalkd_port"])
+  conn, err := beanstalk.Dial("tcp", host)
+
+  if err != nil {
+    log.Info("Beanstalk create connection failed: ", err)
+    panic(err)
+  }
+
+  tubeName := fmt.Sprintf("%s.%s", e.Control["tube_namespace"], e.Control["queue_name"])
+  conn.Tube = beanstalk.Tube { Conn: conn, Name: tubeName }
+  jobBody, err := json.Marshal(e.Payload)
+  if err != nil {
+    log.Info(fmt.Sprintf("* Beanstalkd job payload format invalid: %s", string(jobBody)))
+    panic(err)
```

Comment:
i think the `dispatcher.sendEvent` didn't recover from panic, probably need to update that method if you want to use panic as function break here